sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1229,6 +1229,18 @@ static int sof_card_dai_links_create(struct device *dev,
 		}
 	}
 
+	/*
+	 * BE id fixup
+	 * AMP is always assigned one capture BE ID no matter
+	 * it has capture link or not
+	 * So if be_id is SDW_AMP_DAI_ID + 1, it mean
+	 * the AMP capture BE ID is not reserved. Let's
+	 * adjust it to the proper value before going through
+	 * the other links.
+	 */
+	if (be_id == SDW_AMP_DAI_ID + 1)
+		be_id++;
```

Comment:
@libinyang while the code seems OK, Im not sure if the placement is correct. I feel like you should do this inside the create_sdw_dailink() function

Reply:
@ranj063 I would like close this PR as @plbossart mentioned it is not good to break the existing products.